spring-cloud-aws-autoconfigure/src/test/java/io/awspring/cloud/autoconfigure/cloudmap/integration/EksCloudMapIntegrationTest.java

```diff
@@ -0,0 +1,177 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.awspring.cloud.autoconfigure.cloudmap.integration;
+
+import java.time.Duration;
+import java.util.List;
+
+import io.awspring.cloud.autoconfigure.cloudmap.AwsCloudMapStoreClientCustomizer;
+import io.awspring.cloud.autoconfigure.cloudmap.CloudMapUtils;
+import io.awspring.cloud.autoconfigure.cloudmap.discovery.CloudMapDiscoveryClient;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.testcontainers.containers.localstack.LocalStackContainer;
+import org.testcontainers.junit.jupiter.Container;
+import org.testcontainers.junit.jupiter.Testcontainers;
+import org.testcontainers.utility.DockerImageName;
+import software.amazon.awssdk.core.client.config.ClientOverrideConfiguration;
+import software.amazon.awssdk.http.SdkHttpClient;
+import software.amazon.awssdk.http.apache.ApacheHttpClient;
+import software.amazon.awssdk.services.servicediscovery.ServiceDiscoveryClient;
+
+import org.springframework.boot.BootstrapRegistry;
+import org.springframework.boot.BootstrapRegistryInitializer;
+import org.springframework.boot.SpringApplication;
+import org.springframework.boot.WebApplicationType;
+import org.springframework.boot.autoconfigure.SpringBootApplication;
+import org.springframework.boot.test.system.OutputCaptureExtension;
+import org.springframework.context.ConfigurableApplicationContext;
+
+import static io.awspring.cloud.autoconfigure.cloudmap.CloudMapUtils.EKS;
+
+/**
+ * Integration test for Eks based Cloud Map registration.
+ *
+ * @author Hari Ohm Prasath
+ * @since 3.0
+ */
+@Testcontainers
+@ExtendWith(OutputCaptureExtension.class)
+public class EksCloudMapIntegrationTest {
+
+	// Local stack container with Cloud Map and Ec2 services enabled
+	@Container
+	private static final LocalStackContainer localStackContainer = new LocalStackContainer(
+		DockerImageName.parse("localstack/localstack:1.1.0"))
+		.withServices(LocalStackContainer.EnabledService.named("servicediscovery"), LocalStackContainer.Service.EC2)
+		.withEnv(IntegrationTestUtil.LOCAL_STACK_API, IntegrationTestUtil.LOCAL_STACK_API_KEY)
+		.withReuse(true);
+
+	/**
+	 * Create all the pre-requisites for the test.
+	 */
+	@BeforeAll
+	static void beforeAll() {
+		final String vpcId = IntegrationTestUtil.createVpc(localStackContainer);
+		IntegrationTestUtil.createSubnet(localStackContainer, vpcId);
+		IntegrationTestUtil.createCloudMapResources(localStackContainer, IntegrationTestUtil.TEST_DEFAULT_NAMESPACE
+			, IntegrationTestUtil.TEST_DEFAULT_SERVICE, vpcId);
+	}
+
+	/**
+	 * Register a service with no specification provided in application.properties.
+	 */
+	@Test
+	void registerEksContainerWithCloudMapWithNoSpecification() {
```

Comment:
curios on this.. ECS makes immediate sense as at the pure container layer it is basically just DNS. On the other hand EKS is in k8s, so there's services, endpoints etc.

what's your opinion on registration here clashing or otherwise being redundant with k8s (endpoint, service, etc) or layers over it (istio)? Is this to support multi-clusters? or is the idea to duplicate register for those using raw cloud map?

curious your thinking as I'm a cloudmap noob and missing the insight on this.